### PR TITLE
Add shred tile

### LINKED ISSUE #669: Get Solana Labs bank to shredder integration and outbound TX path working
- [x] Generate shred destinations
- [x] Decide how to handle serialized entry batches larger than 64k
- [x] Store shreds in blockstore

### LINKED ISSUE #706: Merge in and update retransmit tile
Need to switch pubkey validation to from the leader schedule

### PR DESCRIPTION
**Pull request by ptaffet-jump**
Brings in an updated version of the milestone 1.3 shred tile that also calculates shred destination. Needs more testing before merging.
Closes #669 
Closes #706

### CODE REVIEW

File: src/app/fdctl/configure/ethtool.c [hunk 1]
```diff
@@ -170,10 +172,10 @@ check( config_t * const config ) {
     device_read_slaves( config->net.interface, line );
     char * saveptr;
     for( char * token=strtok_r( line, " \t", &saveptr ); token!=NULL; token=strtok_r( NULL, " \t", &saveptr ) ) {
-      CHECK( check_device( token, config->layout.verify_tile_count ) );
+      CHECK( check_device( token, config->layout.verify_tile_count+1U ) );
     }
   } else {
-    CHECK( check_device( config->net.interface, config->layout.verify_tile_count ) );
+    CHECK( check_device( config->net.interface, config->layout.verify_tile_count+1U ) );
```
Comment (mmcgee-jump) [inline src/app/fdctl/configure/ethtool.c:178]: Need to figure out how to set RSS to only go to verify_tile_count right?

File: src/app/fdctl/configure/workspace.c [hunk 2]
```diff
@@ -139,6 +146,20 @@ static void buf( void * pod, char * fmt, void * value, ulong sz, ... ) {
     FD_LOG_ERR(( "failed to initialize workspace" ));
 }
 
+
+
+static inline uchar *
+read_key( char const * key_path,
+          uchar      * key       ) {
+  FILE * key_file = fopen( key_path, "r" );
+  if( FD_UNLIKELY( !key_file ) ) FD_LOG_ERR(( "Opening key file (%s) failed", key_path ));
```
Comment (mmcgee-jump) [inline src/app/fdctl/configure/workspace.c:155]: (%i-%s), errno, strerror( errno )

Comment (mmcgee-jump) [reply src/app/fdctl/configure/workspace.c:155]: Can you make the ENOENT case a special descriptive error message, I feel like it's going to happen frequently.

```C
if( FD_UNLIKELY( errno == ENOENT ) ) FD_LOG_ERR(( "The [consensus.identity_path] in your configuration expects a keyfile at %s but there is no such file. Either update the configuration file to point to your validator identity keypair, or generate a new validator identity key by running `fdctl keygen`", key_path ));
```

File: src/app/fdctl/configure/workspace.c [hunk 3]
```diff
@@ -139,6 +146,20 @@ static void buf( void * pod, char * fmt, void * value, ulong sz, ... ) {
...
+
+  if( FD_UNLIKELY( 1!=fscanf( key_file, "[%hhu", &key[0] ) ) ) FD_LOG_ERR(( "parsing key file failed at pos=0" ));
+  for( ulong i=1UL; i<64UL; i++ ) if( FD_UNLIKELY( 1!=fscanf( key_file, ",%hhu", &key[i] ) ) ) FD_LOG_ERR(( "parsing key file failed at pos=%lu", i ));
```
Comment (mmcgee-jump) [inline src/app/fdctl/configure/workspace.c:158]: Can you scanf the last "]" and also then check for feof to make sure the file is valid.

File: src/app/fdctl/configure/workspace.c [hunk 4]
```diff
@@ -139,6 +146,20 @@ static void buf( void * pod, char * fmt, void * value, ulong sz, ... ) {
...
+  fclose( key_file );
```
Comment (mmcgee-jump) [inline src/app/fdctl/configure/workspace.c:159]: Check result of fclose to not leak the fd

File: src/ballet/bmtree/fd_bmtree.c [hunk 5]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (mmcgee-jump) [inline src/ballet/bmtree/fd_bmtree.c:1]: nit: remove newline

File: src/ballet/wsample/fd_wsample.c [hunk 6]
```diff
@@ -272,6 +272,8 @@ fd_chacha20rng_t * fd_wsample_get_rng( fd_wsample_t * tree ) { return tree->rng;
 void
 fd_wsample_seed_rng( fd_chacha20rng_t * rng,
                      uchar seed[static 32] ) {
+  /* FIXME: Hack, but _init doesn't seem to clear all the state. */
```
Comment (mmcgee-jump) [inline src/ballet/wsample/fd_wsample.c:275]: Prob should fix in rng?

**Comment by ptaffet-jump** [comment]
Some of this has already been committed, all of it needs to be rebased, etc. Closing it and opening a new PR (or several).

**Pull request closed without merging**